src/android/StatusReport.java

```diff
@@ -0,0 +1,80 @@
+package com.microsoft.cordova;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class StatusReport {
+
+    private static final String STATUS_KEY = "status";
+    private static final String LABEL_KEY = "label";
+    private static final String APP_VERSION_KEY = "appVersion";
+    private static final String DEPLOYMENT_KEY_KEY = "deploymentKey";
+    private static final String LAST_VERSION_LABEL_OR_APP_VERSION_KEY = "lastVersionLabelOrAppVersion";
+    private static final String LAST_VERSION_DEPLOYMENT_KEY_KEY = "lastVersionDeploymentKey";
+
+    ReportingStatus status;
+    String label;
+    String appVersion;
+    String deploymentKey;
+    String lastVersionLabelOrAppVersion;
+    String lastVersionDeploymentKey;
+
+    public StatusReport(int status, String label, String appVersion, String deploymentKey, String lastVersionLabelOrAppVersion, String lastVersionDeploymentKey) {
```

Comment:
you may want a `public StatusReport(int status, String label, String appVersion, String deploymentKey)` as well